arangod/Cluster/ClusterInfo.cpp

```diff
@@ -1338,11 +1398,41 @@ void ClusterInfo::loadCurrent() {
       continue;
     }
 
+    std::vector<std::string> colsPath {
+      AgencyCommHelper::path(), "Current", "Collections", databaseName};
     std::vector<std::string> dbPath{
       AgencyCommHelper::path(), "Current", "Databases", databaseName};
     auto databaseSlice = database.second->slice()[0];
+
+    // Database missing in Current
     if (!databaseSlice.hasKey(dbPath)) {
-      newDatabases.erase(databaseName);
+
+      // newShardIds
+      std::shared_ptr<VPackBuilder> db = nullptr;
+      {
+        READ_LOCKER(guard, _currentProt.lock);
+        auto const it = _current.find(databaseName);
+        if (it != _current.end()) {
+          db = it->second;
+        }
+      }
+      std::vector<std::string> colPath {
+        AgencyCommHelper::path(), "Current", "Collections", databaseName};
+
+      if (db != nullptr) {
+        if (db->slice()[0].hasKey(colPath)) {
+          auto const colsSlice = db->slice()[0].get(colPath);
+          if (colsSlice.isObject()) {
+            for (auto const cc : VPackObjectIterator(colsSlice)) {
+              if (cc.value.isObject()) {
+                for (auto const cs : VPackObjectIterator(cc.value)) {
+                  newShardIds.erase(cs.key.copyString());
+                }
+              }
+            }
+          }
+        }
+      }
```

Comment:
Avoiding all unnecessary work, this can be simplified to:
```suggestion
      if (db != nullptr) {
        std::vector<std::string> colPath {
            AgencyCommHelper::path(), "Current", "Collections", databaseName};

        auto colsSlice = db->slice()[0].get(colPath);
        if (colsSlice.isObject()) {
          for (auto const cc : VPackObjectIterator(colsSlice)) {
            if (cc.value.isObject()) {
              for (auto const cs : VPackObjectIterator(cc.value)) {
                newShardIds.erase(cs.key.copyString());
              }
            }
          }
        }
      }
```